public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx

```diff
@@ -289,30 +296,26 @@ function FolderCreationModal({
   const styles = useStyles2(getStyles);
 
   const [title, setTitle] = useState('');
+  const [createFolder] = useNewFolderMutation();
+
   const onSubmit = async () => {
-    const newFolder = await createFolder({ title: title });
-    if (!newFolder.uid) {
-      appEvents.emit(AppEvents.alertError, ['Folder could not be created']);
-      return;
-    }
+    const { data, error } = await createFolder({ title });
 
-    const folder: Folder = { title: newFolder.title, uid: newFolder.uid };
-    onCreate(folder);
-    appEvents.emit(AppEvents.alertSuccess, ['Folder Created', 'OK']);
-  };
+    if (error) {
+      dispatch(notifyApp(createErrorNotification('Failed to create folder')));
```

Comment:
We can use `useAppNotification` instead of dispatching, I believe. It also makes it easier to test with (you can just render `AppNotificationList` in your tests and assert that the toast said the right thing)

Reply:
(pushed this as I made the changes anyway to sanity check myself)